src/examples/using-with-cmake/CMakeLists.txt

```diff
@@ -3,37 +3,77 @@
 #
 # SPDX-License-Identifier: (BSD-3-Clause)
 #------------------------------------------------------------------------------
-# Example that shows how to use an installed instance of Axom in another
-# CMake-based build system.
+# Example that shows how to use an installed instance of Axom 
+# in a CMake-based build system.
+#
+# Required configuration variables are stored in a CMake cache file 
+# called 'host-config.cmake' in the same directory
+#
+# This file defines paths to axom and optionally conduit.
+# It also contains information about the compiler used to build axom.
 #------------------------------------------------------------------------------
 #
 # To build:
 #  mkdir build
 #  cd build
-#  cmake -DAXOM_DIR={axom install path} ../
+#  cmake -C ../host-config.cmake ..
 #  make
 #  ./example
 #
-# Note:
-#  If you use features that leverage external third party libraries 
-#  (ex: Conduit features), the imported CMake targets will handle linking 
-#  these libraries, however you may need to add additional include
-#  include directories to use their headers. 
-# 
 #------------------------------------------------------------------------------
 
 cmake_minimum_required(VERSION 3.8)
 
 project(using_with_cmake)
 
-include("FindAxom.cmake")
+# Simple function to check validity of a path variable, e.g. `FOO_DIR`
+# and a required file under it of the form `${FOO_DIR}/some/subpath/file`
+function(check_var_and_path)
+    set(_variable ${ARGV0})
+    set(_file     ${ARGV1})
+
+    if(NOT DEFINED ${_variable})
+        message(FATAL_ERROR "Required variable '${_variable}' not provided."
+                            " Please supply '${_variable}' to configuration." )
+    endif()
+
+    if(NOT EXISTS ${_file})
+        message (FATAL_ERROR "Path '${_variable}' does not contain required"
+                             " file '${_file}'.")
+    endif()
+endfunction()
```

Comment:
this is handy, but a possible draw back is this looks like cmake vodoo and it's the first thing folks will have to process when trying to learn how to use Axom.

Reply:
Thanks @cyrush -- I removed this function and streamlined these `CMakeLists.txt` files.